packages/xstate-wallet/src/ui/stories/enable-ethereum.stories.tsx

```diff
@@ -17,19 +18,34 @@ const testContext = {
   requestId: 55
 };
 
-if (config.states) {
-  Object.keys(config.states).forEach(state => {
-    const machine = interpret<any, any, any>(
-      ethereumEnableWorkflow(store, messagingService, testContext).withContext(testContext),
-      {
-        devTools: true
-      }
-    ); // start a new interpreted machine for each story
-    machine.onEvent(event => console.log(event.type)).start(state);
-    storiesOf('Workflows / Enable Ethereum', module).add(
-      state.toString(),
-      renderComponentInFrontOfApp(<EnableEthereum current={machine.state} send={machine.send} />)
-    );
-    machine.stop(); // the machine will be stopped before it can be transitioned. This means the console.log on L49 throws a warning that we sent an event to a stopped machine.
-  });
-}
+const windowMetamaskOk = {ethereum: {networkVersion: process.env.CHAIN_NETWORK_ID}};
+const windowNoMetamask = {};
+const windowWrongNetwork = {ethereum: {networkVersion: 3}};
+
+const storyOf = (state, window: any = windowMetamaskOk, name: any = undefined) => {
+  name = name || state.toString();
+  const machine = interpret<any, any, any>(
+    ethereumEnableWorkflow(store, messagingService, testContext).withContext(testContext),
+    {
+      devTools: true
+    }
+  ); // start a new interpreted machine for each story
+  machine.onEvent(event => console.log(event.type)).start(state);
+  storiesOf('Workflows / Enable Ethereum', module).add(
+    name,
+    renderComponentInFrontOfApp(
+      <WindowContext.Provider value={window}>
+        <EnableEthereum current={machine.state} send={machine.send} />
+      </WindowContext.Provider>
```

Comment:
Wondering if you could have used `.withContext` as on L28?

Reply:
I think that's a different sort of context: it's the xstate machine context, rather than the react `useContext` context.

Reply:
Yes you're right.

 I was getting confused with the `.addDecorator` that we use here https://github.com/statechannels/monorepo/blob/master/packages/web3torrent/.storybook/config.js . 

I guess what you have is fine for now but if we wanted to reuse L37-39  ^ this might be an option.